pkg/loader/model_test.go

```diff
@@ -262,3 +265,300 @@ func (s *SQLSuite) TestUpdateMarkSQL(c *check.C) {
+		FieldType: types.FieldType{Tp: mysql.TypeDatetime, Decimal: 0},
+	}
+	colVaue = "2021-09-13 10:10:23"
+	val = genOracleValue(&columnInfo, colVaue)
+	c.Assert(
+		val, check.Equals,
+		"TO_DATE('2021-09-13 10:10:23', 'yyyy-mm-dd hh24:mi:ss')")
+
+	columnInfo = model.ColumnInfo{
+		FieldType: types.FieldType{Tp: mysql.TypeDatetime, Decimal: 6},
+	}
+	colVaue = "2021-09-13 10:10:23.123456"
+	val = genOracleValue(&columnInfo, colVaue)
+	c.Assert(
+		val, check.Equals,
+		"TO_TIMESTAMP('2021-09-13 10:10:23.123456', 'yyyy-mm-dd hh24:mi:ss.ff6')")
+
+	columnInfo = model.ColumnInfo{
+		FieldType: types.FieldType{Tp: mysql.TypeTimestamp, Decimal: 5},
+	}
+	colVaue = "2021-09-13 10:10:23.12345"
+	val = genOracleValue(&columnInfo, colVaue)
+	c.Assert(
+		val, check.Equals,
+		"TO_TIMESTAMP('2021-09-13 10:10:23.12345', 'yyyy-mm-dd hh24:mi:ss.ff5')")
+
+	columnInfo = model.ColumnInfo{
+		FieldType: types.FieldType{Tp: mysql.TypeYear},
+	}
+	colVaue = "2021"
+	val = genOracleValue(&columnInfo, colVaue)
+	c.Assert(
+		val, check.Equals, "2021")
+
+	columnInfo = model.ColumnInfo{
+		FieldType: types.FieldType{Tp: mysql.TypeVarchar},
+	}
+	colVaue = "2021"
+	val = genOracleValue(&columnInfo, colVaue)
+	c.Assert(
+		val, check.Equals, "'2021'")
+
+	columnInfo = model.ColumnInfo{
+		FieldType: types.FieldType{Tp: mysql.TypeDuration},
+	}
+	colVaue = "23:11:59"
+	val = genOracleValue(&columnInfo, colVaue)
+	c.Assert(
+		val, check.Equals, "TO_DATE('23:11:59', 'hh24:mi:ss')")
+
+	var colVaue2 interface{}
+	val = genOracleValue(&columnInfo, colVaue2)
+	c.Assert(
+		val, check.Equals, "NULL")
+}
+
+func (s *SQLSuite) TestOracleDeleteNewValueSQLWithOneUK(c *check.C) {
+	dml := DML{
+		Tp:       InsertDMLType,
+		Database: "db",
+		Table:    "tbl",
+		Values: map[string]interface{}{
+			"ID":   123,
+			"NAME": "pc",
+			"C2":   nil,
+		},
+		info: &tableInfo{
+			columns: []string{"ID", "NAME", "C2"},
+			uniqueKeys: []indexInfo{
+				{
+					name:    "uniq_index_name",
+					columns: []string{"ID"},
+				},
+			},
+		},
+		UpColumnsInfoMap: map[string]*model.ColumnInfo{
+			"ID": {
+				FieldType: types.FieldType{Tp: mysql.TypeInt24}},
+			"NAME": {
+				FieldType: types.FieldType{Tp: mysql.TypeVarString}},
+			"C2": {
+				FieldType: types.FieldType{Tp: mysql.TypeVarString}},
+		},
+	}
+
+	sql := dml.oracleDeleteNewValueSQL()
+	c.Assert(
+		sql, check.Equals,
+		"DELETE FROM db.tbl WHERE ID = 123 AND rownum <=1")
+}
+
+func (s *SQLSuite) TestOracleDeleteNewValueSQLWithMultiUK(c *check.C) {
+	dml := DML{
+		Tp:       InsertDMLType,
+		Database: "db",
+		Table:    "tbl",
+		Values: map[string]interface{}{
+			"ID":   123,
+			"ID2":  "456",
+			"NAME": "pc",
+			"C2":   nil,
+		},
+		info: &tableInfo{
+			columns: []string{"ID", "ID2", "NAME", "C2"},
+			uniqueKeys: []indexInfo{
+				{
+					name:    "uniq_index_name_id2",
+					columns: []string{"ID2"},
+				},
+				{
+					name:    "uniq_index_name_id",
+					columns: []string{"ID"},
+				},
```

Comment:
make sure not picking a unique index with nil value
```suggestion
				{
					name:    "uniq_index_name_id",
					columns: []string{"ID", "C2"},
				},
				{
					name:    "uniq_index_name_id2",
					columns: []string{"ID2"},
				},
```